src/components/Api.js

```diff
@@ -0,0 +1,118 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    // constructor body
+    this._baseUrl = baseUrl;
+    this._authorization = headers.authorization;
+    this._contentType = headers["Content-Type"];
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
```

Comment:
Needs correction: if the response is not ok you should throw an error, so we will fall into catch block